ethcore/src/client/client.rs

```diff
@@ -2199,6 +2207,10 @@ impl IoClient for Client {
 					);
 					if let Err(e) = result {
 						error!(target: "client", "Error importing ancient block: {}", e);
+
+						let mut queued = queued.write();
+						queued.0.clear();
+						queued.1.clear();
```

Comment:
This should work - I believe in the next round it will attempt to download the next batch, which will fail with `UnknownParent`, and then start again from the last successfully imported ancient block. See https://github.com/paritytech/parity-ethereum/blob/master/ethcore/sync/src/block_sync.rs#L522